monitor/main.go

```diff
@@ -461,12 +461,23 @@ func blockProcessor(ctx context.Context, headers <-chan *types.Header, cc *clien
 		transactionCtx := context.Background()
 		for _, txHash := range header.Transactions {
 			receipt, err := cc.Eth.TransactionReceipt(transactionCtx, txHash)
+			
 			if err != nil {
 				return err
 			}
 
 			txLogger := getTxLogger(logger, receipt, header)
-			logTransaction(txLogger)
+			
+			tx, isPending, err := cc.Eth.TransactionByHash(ctx, txHash)
```

Comment:
if you digg a little deeper, you can have a block monitor instead of a header monitor, which will already give you all the transactions, and it would be more efficient.
In fact, i think there's no RPC for the Header only, so the header monitor is already getting the transactions but not parsing them.

With the current solution you're adding N roundtrips and sequential (same applies to receipts though)